src/lib/Libnet/hnls.c

```diff
@@ -663,14 +760,14 @@ get_all_ips(char *msg)
 
 		}
 		if (pIPAddrTable == NULL) {
-			strncpy(msg, "Memory allocation failed for GetIpAddrTable", LOG_BUF_SIZE - 1);
+			strncpy(msg_buf, "Memory allocation failed for GetIpAddrTable", msg_buf_len);
 			return NULL;	
 		}
 	}
 	// Make a second call to GetIpAddrTable to get the
 	// actual data we want
 	if ( (dwRetVal = GetIpAddrTable( pIPAddrTable, &dwSize, 0 )) != NO_ERROR ) { 
-		strncpy(msg, "GetIpAddrTable failed", LOG_BUF_SIZE - 1);
+		strncpy(msg_buf, "GetIpAddrTable failed", msg_buf_len);
```

Comment:
should we be free()ing pIPAddrTable and nodenames before return?

Reply:
yes it is being freed at the bottom of the function, which is the only exit way